diff/print.go

```diff
@@ -4,13 +4,25 @@ import (
 	"bytes"
 	"fmt"
 	"io"
+	"sort"
 	"time"
 
 	"sourcegraph.com/sqs/pbtypes"
 )
 
-// PrintMultiFileDiff prints a multi-file diff in unified diff format.
-func PrintMultiFileDiff(ds []*FileDiff) ([]byte, error) {
+// PrintOptions specifies how to print a diff.
+type PrintOptions struct {
+	Sort bool // sort by filename
+}
+
+// PrintMultiFileDiff prints a multi-file diff in unified diff format with the specified options.
+func PrintMultiFileDiff(ds []*FileDiff, opt *PrintOptions) ([]byte, error) {
```

Comment:
You need to make sure that these call sites can still use the function. I am concerned that it mutates the input argument, which will cause race conditions in:
- https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/frontend/graphqlbackend/repository_comparison.go#L262:17
- https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/pkg/vcs/git/diff_filter.go#L83:22